tiledb/sm/serialization/query.cc

```diff
@@ -504,6 +503,14 @@ Status query_from_capnp(
         "Cannot deserialize; Query opened for " + query_type_str(type) +
         " but got serialized type for " + query_reader.getType().cStr()));
 
+  // Deserialize layout.
```

Comment:
These blocks can be moved back to where they were because we restore the old state in the even that this function fails, is that right?

Reply:
Yeah -- this is that change. The diff here is showing the lines in their original position before #1424